### PR TITLE
database website bulk creation

### PR DESCRIPTION
**Pull request by lukasstreit**
Moved website storage from single files to storing all websites with a bulk create once.

### CODE REVIEW

File: src/storer.ts [hunk 1]
```diff
@@ -166,12 +132,53 @@ export class Storer {
             }
             this.blobService.createBlockBlobFromText(this.container, blobName, compressedBlobContent, (err) => {
                 if(callback) {
-                    callback(err);
+                    return callback(err);
                 }
+                callback();
```
Comment (lgfrbcsgo) [inline src/storer.ts]: Wait, what? Your calling the callback even though it's not defined?

File: src/storer.ts [hunk 2]
```diff
@@ -166,12 +132,53 @@ export class Storer {
...
-
         });
 
         this.blob = undefined;
     }
+
+    public flushDatabase(callback ? : (err? : Error) => void) : void {
+        const websiteInserts = [];
+        const containsInserts = [];
+        this.websites.forEach(website => {
```
Comment (lgfrbcsgo) [inline src/storer.ts]: Why use ugly `.forEach()` here instead of a `for (let website of this.websites)` loop?

File: src/storer.ts [hunk 3]
```diff
@@ -166,12 +132,53 @@ export class Storer {
...
+            let websiteId = Storer.uuid();
+            websiteInserts.push({
+                id: websiteId,
+                url: website.url,
+                blob_url: website.blobUrl
+            });
+            website.occurences.forEach(occ => {
```
Comment (lgfrbcsgo) [inline src/storer.ts]: Still don't like `.forEach()` ...

File: src/storer.ts (hunk 1 above)
Comment (lukasstreit) [inline src/storer.ts]: fair point.

File: src/storer.ts (hunk 2 above)
Comment (lukasstreit) [inline src/storer.ts]: [http://thejsguy.com/2016/07/30/javascript-for-loop-vs-array-foreach.html](http://thejsguy.com/2016/07/30/javascript-for-loop-vs-array-foreach.html)
Don't mess with my forEach loops :heart:

Comment (lgfrbcsgo) [inline src/storer.ts]: Ok, fair point. I won't abandon my `for` loops though.

**Pull request merged**